### PR TITLE
Add Terraform

### PR DESCRIPTION
**Pull request by shuaibiyy**
https://github.com/shuaibiyy/awesome-terraform

A curated list of resources on HashiCorp's Terraform. Terraform enables and simplifies the practice of infrastructure-as-code.

# By submitting this pull request I confirm I've read and complied with the below requirements.

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title.<br>For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with these requirements:
	- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
	- It's the result of hard work and the best I could possibly produce.
	- Non-generated Markdown file in a GitHub repo.
	- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
	- Not a duplicate.
	- Only has awesome items. Awesome lists are curations of the best, not everything.
	- Includes a project logo/illustration whenever possible.
		- Placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
		- The image should link to the project website or any relevant website.
		- The image should be high-DPI. Set it to maximum half the width of the original image.
	- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
	- Has the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- Has a Table of Contents section.
		- Should be named `Contents`, not `Table of Contents`.
		- Should be the first section in the list.
	- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
		- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
		- If you use a license badge, it should be SVG, not PNG.
	- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
		- The file should be named `contributing.md`. Casing is up to you.
	- Has consistent formatting and proper spelling/grammar.
		- Each link description starts with an uppercase character and ends with a period.<br>Example: `- [AVA](…) - JavaScript test runner.`
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
	- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Go to the top and read it again.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -235,6 +235,7 @@
 - [Serverless Framework](https://github.com/JustServerless/awesome-serverless)
 - [Apache Wicket](https://github.com/PhantomYdn/awesome-wicket) - Java web app framework.
 - [Vert.x](https://github.com/vert-x3/vertx-awesome) - Toolkit for building reactive apps on the JVM.
+- [Terraform](https://github.com/shuaibiyy/awesome-terraform)
```
Comment (kevva) [inline readme.md]: Missing description.

**Comment by kevva** [comment]
It's missing the `awesome` badge. Many of the links are missing descriptions too.

**Comment by shuaibiyy** [comment]
@kevva Thanks for taking a look. I've added a description and awesome badge.

> Many of the links are missing descriptions too.

Are you referring to the blog posts? I think the titles are descriptive, no?

**Comment by kevva** [comment]
> Are you referring to the blog posts? I think the titles are descriptive, no?

Most of them, yes. I'll let @sindresorhus decide though. Also, a Google search gives me a lot of missing articles linked in the list.

**Comment by sindresorhus** [comment]
Looks pretty good. Some feedback:

---

> **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)

:arrow_up:

---

The items in the `Contents` section is not ordered the same as the actual sections.

---

> Drop all the `A` / `An` prefixes in the descriptions.

:arrow_up:

---

No point in having `Terraform module` part of the description. That's implied. (In Community Modules)

Also moot to prefix the items in the Community section with Terraform.

---

It would also be useful with some personal comments about the talks/blogposts if you have listened/read any. Example: https://github.com/sindresorhus/awesome-scifi#science-fiction

---

The talks and podcasts items could use a description/summary of what they're about. For example, `Terraform w/ Lee Trout`, I don't know who that is, so the description is not very useful. `Terraforming the Composable World` - Ok, but I don't know whether that's intersting to me, need more info to decide whether it's worth my time.

**Comment by shuaibiyy** [comment]
@sindresorhus Thanks for the feedback. I'll make the improvements.

**Comment by sindresorhus** [comment]
@shuaibiyy ping

**Comment by sindresorhus** [comment]
ping2

**Comment by shuaibiyy** [comment]
Thanks for the reminder :) Shall make the improvements this weekend!

**Comment by shuaibiyy** [comment]
@sindresorhus Improvements have been made.

**Comment by sindresorhus** [comment]
Very good. Just one minor thing left:

> Drop all the A / An prefixes in the descriptions.
⬆️

Not done. For example, `A module to provide a VPC in AWS.`, could be improved to `Provide a VPC in AWS.`.

**Comment by shuaibiyy** [comment]
@sindresorhus Done.. I think :)

**Comment by sindresorhus** [comment]
> Done.. I think :)

Nope, still many descriptions with `A`/`An` prefix.

**Comment by shuaibiyy** [comment]
@sindresorhus How about now?

**Comment by sindresorhus** [comment]
Great! Thanks for your patience :)

**Comment by sindresorhus** [comment]
Tweeted: https://twitter.com/awesome__re/status/939983367872933888

**Pull request merged**